cli/azd/pkg/tools/maven/maven.go

```diff
@@ -212,6 +213,30 @@ func (cli *Cli) ResolveDependencies(ctx context.Context, projectPath string) err
 	return nil
 }
 
+var ErrPropertyNotFound = errors.New("property not found")
+
+func (cli *Cli) GetProperty(ctx context.Context, propertyPath string, projectPath string) (string, error) {
```

Comment:
can you add unit tests for this new function?

Reply:
I do believe we're fine without tests here. The isolated in-unit ("azd") code is largely trivial.

The bigger concerns would be integration in nature, as we're very susceptible to external changes. For example, if maven updates the invocation behavior, a unit test wouldn't catch that, and there would be more maintenance burden to continue updating those expectations that would be just as easy to be tested while making the changes.